api/pkg/apis/v1alpha1/managers/activations/activations-manager.go

```diff
@@ -134,6 +134,28 @@ func (m *ActivationsManager) UpsertState(ctx context.Context, name string, state
 	}
 	state.ObjectMeta.FixNames(name)
 
+	getRequest := states.GetRequest{
+		ID: name,
+		Metadata: map[string]interface{}{
+			"version":   "v1",
+			"group":     model.WorkflowGroup,
+			"resource":  "activations",
+			"namespace": state.ObjectMeta.Namespace,
+			"kind":      "Activation",
+		},
+	}
+	var entry states.StateEntry
+	entry, err = m.StateProvider.Get(ctx, getRequest)
+	if err == nil {
+		// preserve system annotations for existing object
+		itemState, err := getActivationState(entry.Body)
+		if err != nil {
+			log.ErrorfCtx(ctx, "Failed to convert to activation state for %s in namespace %s: %v", name, state.ObjectMeta.Namespace, err)
+			return err
+		}
+		state.ObjectMeta.PreserveSystemMetadata(itemState.ObjectMeta)
+	}
+
```

Comment:
Why don't you use "GetState" interface? In addition, there is another call in the ValidateCreateOrUpdate to get previous state which seems duplicate.

Reply:
updated.